File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectProductaanvraag.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import org.apache.commons.lang3.StringUtils;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;

public class ZaakobjectProductaanvraag extends Zaakobject {

    public static final String OBJECT_TYPE_OVERIGE = "ProductAanvraag";

    /**
     * No-arg constructor for JSONB deserialization
     */
    public ZaakobjectProductaanvraag() {
    }

    /**
     * Constructor with required attributes
     */
    public ZaakobjectProductaanvraag(final URI zaak, final URI productaanvraag) {
        super(zaak, productaanvraag, ObjectTypeEnum.OVERIGE);
        setObjectTypeOverige(OBJECT_TYPE_OVERIGE);
    }

    @Override
    public String getWaarde() {
        return StringUtils.substringAfterLast(getObject().getPath(), "/");
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectProductaanvraag.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI
import java.util.UUID

class ZaakobjectProductaanvraag : Zaakobject {
    companion object {
        const val OBJECT_TYPE_OVERIGE_PRODUCTAANVRAAG = "ProductAanvraag"
    }

    /**
     * No-arg constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with required attributes
     */
    constructor(zaak: URI, productaanvraag: URI?, url: URI, uuid: UUID) :
        super(zaak, productaanvraag, ObjectTypeEnum.OVERIGE, url, uuid) {
        objectTypeOverige = OBJECT_TYPE_OVERIGE_PRODUCTAANVRAAG
    }

    override val waarde: String?
        get() = `object`?.path?.substringAfterLast("/")
}
